### PR TITLE
[CDAP-14079] Rename and fix flaky test

### PR DESCRIPTION
**Pull request by anew**
- rename testStartProgramWithDisabledRuntimeArgs() to testStartProgramWithDisabledProfile()
- fix the way it waits for the workflow to stop

### CODE REVIEW

File: cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/ProgramLifecycleHttpHandlerTest.java [hunk 1]
```diff
@@ -136,6 +136,9 @@
   private static final String STOPPED = "STOPPED";
   private static final String RUNNING = "RUNNING";
 
+//  @Rule // 120 seconds max per method tested
+//  public Timeout globalTimeout = new Timeout(5 * 1000);
```
Comment (yaojiefeng) [inline cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/ProgramLifecycleHttpHandlerTest.java]: What is the use of this commented timeout? Can we just remove it?

**Review by yaojiefeng — APPROVED** [review]
just one minor comment, rest LGTM

File: cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/ProgramLifecycleHttpHandlerTest.java [hunk 2]
```diff
@@ -136,6 +136,9 @@
   private static final String STOPPED = "STOPPED";
   private static final String RUNNING = "RUNNING";
 
+//  @Rule // 120 seconds max per method tested
```
Comment (anwar6953) [inline cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/ProgramLifecycleHttpHandlerTest.java]: remove?

File: cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/ProgramLifecycleHttpHandlerTest.java (hunk 1 above)
Comment (anew) [reply cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/ProgramLifecycleHttpHandlerTest.java]: oh yeah that's a relic of something I tested out. will remove.

File: cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/ProgramLifecycleHttpHandlerTest.java (hunk 2 above)
Comment (anew) [reply cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/ProgramLifecycleHttpHandlerTest.java]: oh yeah that's a relic of something I tested out. will remove.

**Comment by anew** [comment]
I am merging this without waiting for the build because it has no risk of introducing new failures. I have run the test on my dev machine.

**Pull request merged**